src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/ThreadPoolMetricsCollector.java

```diff
@@ -52,6 +57,24 @@ public void collectMetrics(long startTime) {
 
         while (statsIterator.hasNext()) {
             Stats stats = statsIterator.next();
+            long rejectionDelta = 0;
+            String name = stats.getName();
+            if (statsRecordMap.containsKey(name)) {
+                ThreadPoolStatsRecord lastRecord = statsRecordMap.get(name);
+                // if the timestamp in previous record is greater than 15s (3 * intervals),
+                // then the scheduler might hang or freeze due to long GC etc. We simply drop
+                // previous record here and set rejectionDelta to 0.
+                if (startTime - lastRecord.getTimestamp() <= SAMPLING_TIME_INTERVAL * 3) {
+                    rejectionDelta = stats.getRejected() - lastRecord.getRejected();
+                    // we might not run into this as rejection is a LongAdder which never decrement its count.
+                    // regardless, let's set it to 0 to be safe.
+                    if (rejectionDelta < 0) {
+                        rejectionDelta = 0;
+                    }
+                }
+            }
+            statsRecordMap.put(name, new ThreadPoolStatsRecord(startTime, stats.getRejected()));
```

Comment:
Do we need to clear the `statsRecordMap` map anywhere ?

Reply:
We don't need to clear the map because there are only a fixed amount of types of queue in thread pool and we always compare the timestamp before writing into the map.